gravitee-kubernetes-client/src/main/java/io/gravitee/kubernetes/client/KubernetesClientV1Impl.java

```diff
@@ -179,26 +153,283 @@ public KubernetesClientV1Impl(Vertx vertx, KubernetesConfig kubernetesConfig) {
+        }
+
+        return promise.future();
+    }
+
+    @Override
+    public Future<Void> stopAll() {
+        watchMap.forEach((k, v) -> v.stopped = true);
+        return Future.succeededFuture();
+    }
+
+    private <T extends Event> Observable<T> fetchEvents(
+        KubeResource resource,
+        String lastResourceVersion,
+        String fieldSelector,
+        String location,
+        Class<T> type
+    ) {
+        WebSocketConnectOptions webSocketConnectOptions = new WebSocketConnectOptions();
+        webSocketConnectOptions.setURI(watcherUrlPath(resource.namespace, lastResourceVersion, fieldSelector, type));
+        webSocketConnectOptions.setHost(config.getApiServerHost());
+        webSocketConnectOptions.setPort(config.getApiServerPort());
+        webSocketConnectOptions.setSsl(true);
+        webSocketConnectOptions.addHeader(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON);
+        webSocketConnectOptions.addHeader(HttpHeaders.AUTHORIZATION, "Bearer " + config.getAccessToken());
+
+        return httpClient
+            .rxWebSocket(webSocketConnectOptions)
+            .flatMapObservable(
+                websocket ->
+                    websocket
+                        .toObservable()
+                        .flatMap(
+                            response -> {
+                                if (watchMap.get(location).stopped) {
+                                    LOGGER.info("Stop Kubernetes resource watcher {}.", location);
+                                    websocket.close();
+                                    return Observable.empty();
+                                }
+
+                                T item = response.toJsonObject().mapTo(type);
+                                // TODO - Kamiel - 12/10/2021: to be replaced with proper operation
+                                if (
+                                    item.getType().equals(KubernetesEventType.MODIFIED.name()) ||
+                                    item.getType().equals(KubernetesEventType.ADDED.name())
+                                ) {
+                                    JsonObject json = response.toJsonObject();
+                                    JsonObject object = json.getJsonObject("object");
+                                    JsonObject metadata = object.getJsonObject("metadata");
+                                    String name = metadata.getString("name");
+
+                                    if (item instanceof ConfigMapEvent) {
+                                        get(String.format("kube://%s/configmap/%s", resource.namespace, name), ConfigMap.class)
+                                            .subscribe(((ConfigMapEvent) item)::setData, Throwable::printStackTrace);
+                                    } else if (item instanceof SecretEvent) {
+                                        get(String.format("kube://%s/secret/%s", resource.namespace, name), Secret.class)
```

Comment:
Why do we need to make a get call to retrieve the secret ? It should be present in the event sent by kubernetes right ?

Also there is no guaranty that the get will be completed before the item is pushed to the observable.